### PR TITLE
EDGAR release 24.2.u1

### PR DESCRIPTION
**Pull request by hermfischer-wf**
#### Reason for change
Updates for EDGAR release 24.2.u1 (deployment July 31st)

#### Description of change
Updates to fee exhibit validation
Updates for R file text block rendering
Ixviewer minor fixes

#### Steps to Test

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/EFM/Filing.py [hunk 1]
```diff
@@ -2400,16 +2400,17 @@ def getEvalFunctionStringAndArgs(sev, functionName, argumentsKey="function-argum
...
-                            found = False
-                            for f in modelXbrl.factsByDimMemQname(axisQN):
-                                if ftName(f) == name:
-                                    if validation.endswith("value") and not f.xValue in value:
-                                        continue
-                                    found = True
-                                    break
-                            if not found:
-                                sevMessage(sev, subType=submissionType, modelObject=None, tag=ftName(name), axis=axisQN, value=value)
+                        if not sev.get("if-axis-exist") or (sev.get("if-axis-exist") and modelXbrl.factsByDimMemQname(axisQN)):
```
Comment (brettkail-wk) [inline arelle/plugin/validate/EFM/Filing.py:2403]: The `sev.get("if-axis-exist")` seems redundant:
```
if not sev.get("if-axis-exist") or modelXbrl.factsByDimMemQname(axisQN):
```

File: arelle/plugin/validate/EFM/Filing.py [hunk 2]
```diff
@@ -2741,14 +2742,12 @@ def addCoverFactRecord(facts):
...
                 jsonObjType = "fee"
-                messageCode = "EFM.feeFacts"
                 testEnvJsonFile = val.params.get("saveFeeFacts")
             else:
                 jsonObjType = "cover"
-                messageCode = "EFM.coverFacts"
                 testEnvJsonFile = val.params.get("saveCoverFacts")
             modelXbrl.log("INFO-RESULT",
-                          messageCode,
+                          "EFM.{}Facts".format(jsonObjType),
```
Comment (brettkail-wk) [inline arelle/plugin/validate/EFM/Filing.py:2750]: This reverts #1221 and should be undone.

File: arelle/plugin/validate/EFM/Filing.py [hunk 3]
```diff
@@ -3902,7 +3901,7 @@ def r6facts():
             elif dqcRuleName == "DQC.US.0036" and hasDocPerEndDateFact:
                 for id, rule in dqcRule["rules"].items():
                     for f in modelXbrl.factsByLocalName.get(rule["name"],()):
-                        if f.context is not None and f.xValid >= VALID and abs((f.xValue + ONE_DAY - f.context.endDatetime).days) > 1: # was 3
+                        if f.context is not None and abs((f.xValue + ONE_DAY - f.context.endDatetime).days) > 1: # was 3
```
Comment (brettkail-wk) [inline arelle/plugin/validate/EFM/Filing.py:3904]: This reverts #1276 and should be undone.

File: arelle/plugin/validate/EFM/Filing.py [hunk 4]
```diff
@@ -4932,11 +4931,7 @@ def getMonetaryDescendants(fromConcept, elr, result, visited=None):
             modelXbrl.warning(f"{dqcRuleName}.{id}",
                               f"Validation was unable to complete rule {dqcRuleName} due to an internal error.  This is not considered an error in the filing.",
                               modelObject=modelXbrl)
-            modelXbrl.debug(
-                "arelle:dqcrtException",
-                _("An unexpected exception occurred in DQCRT"),
-                traceback=traceback.format_exception(*sys.exc_info())
-            )
+            modelXbrl.debug("arelle:dqcrtException", _("DQCRT Exception traceback: {}").format(traceback.format_exception(*sys.exc_info())))
```
Comment (brettkail-wk) [inline arelle/plugin/validate/EFM/Filing.py:4934]: This reverts #1204 again (#1256).

File: arelle/plugin/validate/EFM/config.xml [hunk 5]
```diff
@@ -203,45 +205,6 @@
      standardTaxonomyDatabase="efmStandardTaxonomies.db"
      standardTaxonomyUrlPattern="http://www.fasb.org|http://xbrl.sec.gov|http://www.xbrl.org|http://taxonomies.xbrl.us|http://xbrl.us/us-gaap|http://xbrl.ici.org/rr|http://xbrl.ifrs.org/taxonomy"
      />
-  <DisclosureSystem
```
Comment (brettkail-wk) [inline arelle/plugin/validate/EFM/config.xml:206]: This reverts #1225.

**Review by brettkail-wk — DISMISSED** [review]

File: arelle/plugin/validate/EFM/Filing.py (hunk 2 above)
Comment (hermfischer-wf) [reply arelle/plugin/validate/EFM/Filing.py:2750]: added in messageCodes=("EFM.feeFacts", "EFM.coverFacts") which is used in the other places where dynamic constructs generate multiple message codes.

File: arelle/plugin/validate/EFM/Filing.py (hunk 3 above)
Comment (hermfischer-wf) [reply arelle/plugin/validate/EFM/Filing.py:3904]: Change added back.

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**